functions/Invoke-DbaDiagnosticQuery.ps1

```diff
@@ -0,0 +1,278 @@
+									DatabaseName = $null
+									Notes = "Empty Result for this Query"
+									Result = $null
+								}
+								Write-Message -Level Verbose -Message ("Empty result for Query {0} - {1} - {2}" -f $scriptpart.QueryNr, $scriptpart.QueryName, $scriptpart.Description)
+							}
+						}
+						catch {
+							Write-Message -Level Verbose -Message ('Some error has occured on Server: {0} - Script: {1}, result unavailable' -f $instance, $scriptpart.QueryName)
+						}
+						if ($result) {
+							
+							[pscustomobject]@{
+								ComputerName = $server.NetName
+								InstanceName = $server.ServiceName
+								SqlInstance = $server.DomainInstanceName
+								Number = $scriptpart.QueryNr
+								Name = $scriptpart.QueryName
+								Description = $scriptpart.Description
+								DatabaseSpecific = $scriptpart.DBSpecific
+								DatabaseName = $null
+								Notes = $null
+								Result = $result | Select-Object * -ExcludeProperty RowError, RowState, Table, ItemArray, HasErrors
+							}
+						}
+					}
+				}
+				elseif ($scriptpart.DatabaseSpecific -and !$instanceOnly) {
+					foreach ($database in $databases) {
+						if ($PSCmdlet.ShouldProcess(('{0} ({1})' -f $instance, $database.name), $scriptpart.QueryName)) {
+							#if (!$silent) { Write-Progress -Id 0 -Activity "Running diagnostic queries on SQL Server" -Status ("Instance {0} of {1}" -f $servercounter, $sqlservers.count) -CurrentOperation $instance -PercentComplete (($servercounter / $sqlServers.count) * 100) }
+							if (!$silent) { Write-Progress -Id 1 -ParentId 0 -Activity "Collecting diagnostic query data from $database on $instance" -Status ('Processing {0} of {1}' -f $counter, $scriptcount) -CurrentOperation $scriptpart.QueryName -PercentComplete (($Counter / $scriptcount) * 100) }
+							Write-Message -Level Output -Message "Collecting diagnostic query data from $database for $($scriptpart.QueryName) on $instance"
+							try {
+								$result = Invoke-Sqlcmd2 -ServerInstance $server -Database $database.Name -Query $scriptpart.Text -ErrorAction Stop
+								if (!$result) {
+									$result = [pscustomobject]@{
+										ComputerName = $server.NetName
+										InstanceName = $server.ServiceName
+										SqlInstance = $server.DomainInstanceName
+										Number = $scriptpart.QueryNr
+										Name = $scriptpart.QueryName
+										Description = $scriptpart.Description
+										DatabaseSpecific = $scriptpart.DBSpecific
+										DatabaseName = $null
+										Notes = "Empty Result for this Query"
+										Result = $null
+									}
+									Write-Message -Level Verbose -Message ("Empty result for Query {0} - {1} - {2}" -f $scriptpart.QueryNr, $scriptpart.QueryName, $scriptpart.Description)
+								}
+							}
+							catch {
+								Write-Message -Level Verbose -Message ('Some error has occured on Server: {0} - Script: {1} - Database: {2}, result will not be saved' -f $instance, $scriptpart.QueryName, $database.Name)
```

Comment:
Consider adding the `-Target` parameter, either containing the script or the instance
Consider adding the `-ErrorRecord $_` parameter, to help with debugging and ensure expected unattended behavior